src/persist/src/file.rs

```diff
@@ -359,6 +349,24 @@ impl Blob for FileBlob {
     }
 }
 
+fn file_storage_lock(lockfile_path: &Path, new_lock: LockInfo) -> Result<File, Error> {
+    let mut lockfile = OpenOptions::new()
+        .read(true)
+        .write(true)
+        .create(true)
+        .open(&lockfile_path)?;
+    if let Err(existing) = new_lock.is_reentrant_for(&mut lockfile)? {
```

Comment:
I might be misreading but it seems like `is_reentrant_for` can also return a `Ok(Err)` that isn't handled here?